centreon/packages/ui/src/components/Menu/useMenu.tsx

```diff
@@ -0,0 +1,79 @@
+import { ReactElement, ReactNode, useRef } from 'react';
+
+import {
+  atom,
+  createStore,
+  useAtom,
+  Provider,
+  useStore,
+  useAtomValue
+} from 'jotai';
+import { useHydrateAtoms } from 'jotai/utils';
+
+/** state */
+
+const isMenuOpenAtom = atom(false);
+const anchorElAtom = atom<null | HTMLElement>(null);
+const onOpenAtom = atom<(() => void) | null>(null);
+const onCloseAtom = atom<(() => void) | null>(null);
+
+/** provider */
+
+type MenuProviderProps = {
+  children: ReactNode;
+  initialIsOpen?: boolean;
+  onClose?: () => void;
+  onOpen?: () => void;
+};
+
+const MenuProvider = ({
+  children,
+  initialIsOpen,
+  onOpen,
+  onClose
+}: MenuProviderProps): ReactElement => {
+  const menuStore = useRef(createStore()).current;
+
+  useHydrateAtoms(
```

Comment:
Why using this hook? We are not in a SSR environment

Reply:
as mentioned on Slack, is according to jotai docs

Reply:
You can just use this documentation: https://jotai.org/docs/core/store
I mean use the get and set function from the store

Reply:
i'm afraid the use case is different, the goal is not to create a singleton store, but multiple stores for each instance of a Menu component
→ https://jotai.org/docs/guides/initialize-atom-on-render